telegram/ext/_dispatcher.py

```diff
@@ -631,6 +652,78 @@ def remove_handler(self, handler: Handler, group: int = DEFAULT_GROUP) -> None:
             if not self.handlers[group]:
                 del self.handlers[group]
 
+    def drop_chat_data(self, chat_id: int = None, all_empty_entries: bool = False) -> None:
+        """Used for deleting a key from the :attr:`chat_data`.
+
+        .. versionadded:: 14.0
+
+        Args:
+            chat_id (:obj:`int`, optional): The chat id to delete from the persistence. The entry
+                will be deleted even if it is not empty. Mutually exclusive with
+                :paramref:`all_empty_entries`.
+            all_empty_entries (:obj:`bool`, optional): If :obj:`True`, all empty entries from
+                :attr:`chat_data` will be deleted. Mutually exclusive with :paramref:`chat_id`.
+
+        Raises:
+            :exc:`ValueError`: When both :paramref:`chat_id` and :paramref:`all_empty_entries` are
+                provided, or none of them are passed.
```

Comment:
or chat_id is not present. Is that desired, though? my first instinct would be to debug log & pass

Reply:
it would be more user-friendly if we detected that first directly isn't it

Reply:
if we don't raise an exception, then the outcome is always that `chat_id not in dispatcher.chat_data` after calling the method. Otherwise the outcome depends on `chat_id in dispatcher.chat_data` … Personally I find the first one more user friendly …